Correction to AirflowNetwork:Multizone:Surface Handling of People Index

RKStrand opened the pull request:
Pull request overview
---------------------

- Fixes #11027 

### Description of the purpose of this PR

For situations where a user specified AirflowNetwork:Multizone:Surface objects with CEN15251 (or ASH55) venting controls but the AirflowNetwork:Multizone:Zone object used some other venting control, no link was made to the correct people statement.  As a result, no venting was ever done for this situation which was incorrect.  Now, the code links up the proper zone and then links up to the correct People statement and venting occurs as specified by the user via a new subroutine which handles both ASH55 and CEN15251 for both the :Surface and :Zone objects.  Two IDFs were included with the defect.  One did not work because it met the condition shown above.  The second had the :Zone object set to CEN15251 which then did allow venting.  There were clear differences in the outputs.  After the fix, both files produce exactly the same results as expected and these match the results of the second input file with :Zone set to CEN15251.  No differences were noted in the ci results as anticipated.  A minor documentation edit and a unit test are included.

### Pull Request Author

<!-- Add to this list or remove from it as applicable.  This is a simple templated set of guidelines. -->

 - [X] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [X] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [X] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer

<!-- This will not be exhaustively relevant to every PR. -->

 - [x] Perform a Code Review on GitHub
 - [x] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [x] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [x] CI status: all green or justified
 - [x] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

RKStrand committed:
Solution for Surface Level Indices not being set

The code for AFN was not setting the zone number properly for surfaces within the AFN.  Code existed for zone input to do this and link to a people statement for ASHRAE55 and CEN15251 adaptive modeling.  The correction included getting all zone and surfaces to use the same (new) subroutine, simplifying the code within the main get routine and relegating the details of linking surfaces, zones, and people statements to the new subroutine.

RKStrand committed:
Unit test and doc tweak

Slight update to the new subroutine plus the addition of a unit test and a small clarification for the IO Reference section on AFN.

mjwitte commented on src/EnergyPlus/AirflowNetwork/src/Solver.cpp:
```diff
@@ -2500,31 +2500,12 @@ namespace AirflowNetwork {
-                                           " requires a people object with respective model calculations.");
-                    }
+                    int mzZoneDataPtr;
+                    Solver::get_people_index(MultizoneZoneData(i).CEN15251PeopleInd, mzZoneDataPtr, VentControlType::CEN15251, false, i, ErrorsFound);
```
This works, but I'm puzzled why there are some many function parameters. I would expect this to be an `int` function:

`MultizoneZoneData(i).CEN15251PeopleInd =  Solver::get_people_index( MultizoneZoneData(i).ZoneNum, VentControlType::CEN15251, ErrorsFound);`

mjwitte commented on src/EnergyPlus/AirflowNetwork/src/Solver.cpp:
```diff
@@ -3322,9 +3303,20 @@ namespace AirflowNetwork {
                 case VentControlType::ASH55:
-                case VentControlType::CEN15251:
+                case VentControlType::CEN15251: {
+                    int peopleIndex;
```
```
int mzPtr = MultizoneSurfaceData(i).ZonePtr
MultizoneZoneData(mzPtr).CEN15251PeopleInd =  Solver::get_people_index(mzPtr , VentControlType::CEN15251, ErrorsFound);
```

With a similar block for case `VentControlType::ASH55`

RKStrand replied:
I see your point.  There is a lot more complication than seems warranted.  The zone pointer should be available already so sending that should avoid having to have the first block of code.  I have often avoided setting one of the passed variables to a function like this one and made those general subroutines, but I guess there is nothing wrong with making this a function and still setting the error flag that is passed.  I'll implement those changes and then mark this for re-review when I'm done.  Thanks!

RKStrand replied:
@mjwitte Ok--I have reworked this so that it's a function and the function is much simpler/leaner.  Just made the commit of code that still does what it needs to do.  Also modified the unit test accordingly.  Hopefully this addresses your concern.

mjwitte reviewed (commented):
This works as-is, but the new function could be much simpler, I think.

RKStrand committed:
Changes to Solution Based on Review

Simplified solution by making the subroutine a function and narrowing down what the function does.

RKStrand committed:
One more AFN iteration...

One more simplification/correction.

mjwitte commented:
> @RKStrand So the latest changes no longer fix the defect file. Sorry I suggested simplifying the function. If you want to revert the last two commits and go back to your first solution, that's fine with me.

Oh, wait, I may have run that with the wrong exe - hold on.

mjwitte reviewed (approved):
@RKStrand All good. Thanks! Merging.
